// n8n-nodes-webpage-content-extractor/dist/nodes/Readability/Readability.node.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Readability = void 0;
const readability_1 = require("@mozilla/readability");
const jsdom_1 = require("jsdom");
class Readability {
    constructor() {
        this.description = {
            displayName: 'Readability',
            name: 'readability',
            icon: 'file:readability.svg',
            group: ['transform'],
            version: 1,
            description: '',
            defaults: {
                name: 'Readability',
            },
            inputs: ['main'],
            outputs: ['main'],
            properties: [
                {
                    displayName: 'URL',
                    name: 'url',
                    type: 'string',
                    required: true,
                    default: '',
                    placeholder: 'https://',
                    description: 'URL of a page from which you want to extract the main contents.',
                },
            ],
        };
    }
    async execute() {
        const items = this.getInputData();
        const output = [];
        for (let i = 0; i < items.length; i++) {
            const url = this.getNodeParameter('url', i);
            const textBody = await this.helpers.httpRequest({
                url: url,
            });
            const doc = new jsdom_1.JSDOM(textBody);
            const article = new readability_1.Readability(doc.window.document).parse();
            if (!article) {
                output.push({});
                continue;
            }
            output.push({
                json: {
                    excerpt: article.excerpt,
                    siteName: article.siteName,
                    length: article.length,
                    textContent: article.textContent,
                    title: article.title,
                    language: article.lang,
                }
            });
        }
        return [this.helpers.returnJsonArray(output)];
    }
}
exports.Readability = Readability;
